apps/cowswap-frontend/src/modules/fiatAmount/services/fetchCurrencyFiatPrice.ts

```diff
@@ -0,0 +1,49 @@
+import { Token } from '@uniswap/sdk-core'
+
+import { LONG_PRECISION } from 'legacy/constants'
+
+import {
+  COINGECKO_RATE_LIMIT_TIMEOUT,
+  CoingeckoRateLimitError,
+  getCoingeckoPrice,
+  UnsupporedCoingeckoPlatformError,
+} from '../apis/getCoingeckoPrice'
+import { getCowProtocolFiatPrice } from '../apis/getCowProtocolFiatPrice'
+
+let coingeckoRateLimitHitTimestamp: null | number = null
+
+/**
+ * Fetches fiat price for a given currency from coingecko or CowProtocol
+ * CowProtocol is used as a fallback
+ * When coingecko rate limit is hit, CowProtocol will be used for 1 minute
+ */
+export function fetchCurrencyFiatPrice(currency: Token, usdcPrice$: Promise<number | null>): Promise<number | null> {
+  const shouldSkipCoingecko =
+    !!coingeckoRateLimitHitTimestamp && Date.now() - coingeckoRateLimitHitTimestamp < COINGECKO_RATE_LIMIT_TIMEOUT
+
+  if (coingeckoRateLimitHitTimestamp && !shouldSkipCoingecko) {
+    coingeckoRateLimitHitTimestamp = null
+  }
```

Comment:
Nitpick: create a short helper fn to encapsulate this away, which updates the cached value `getShouldSkipCoingecko()` or similar

Reply:
Thanks! Fixed